calicoctl/calico_ctl/pool.py

```diff
@@ -122,19 +122,58 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
-    for cidr in cidrs:
+    current_pools = client.get_ip_pools(version)
+    current_pool_nets = [IPNetwork(pool.cidr) for pool in current_pools]
+
+    new_pools = []
+
+    for new_cidr in cidrs:
+        new_cidr_net = IPNetwork(new_cidr)
+
+        # Check if new pool overlaps with any existing pool
+        overlapping_old_cidr = _check_if_cidr_overlaps_cidrs(new_cidr,
+                                   [net.cidr for net in current_pool_nets])
```

Comment:
Isn't `[net.cidr for net in current_pool_nets]` just the same as `current_pool_nets`?